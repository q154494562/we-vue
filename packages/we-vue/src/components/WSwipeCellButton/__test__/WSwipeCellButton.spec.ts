import { shallowMount } from '@vue/test-utils'
import SwipeCellButton from '../'

describe('cell-swipe-button', () => {
  test('create', () => {
    let wrapper = shallowMount(SwipeCellButton, {
      propsData: {},
    })

    expect(wrapper.name()).toBe('w-cell-swipe-button')
    expect(wrapper.classes()).toContain('weui-swiped-btn')

    expect(wrapper.html()).toMatchSnapshot()

    // create with type
    wrapper = shallowMount(SwipeCellButton, {
      propsData: {
        type: 'test',
      },
    })

    expect(wrapper.classes()).toContain('weui-swiped-btn_test')

    wrapper = shallowMount(SwipeCellButton, {
      slots: {
        default: 'test',
      },
    })

    expect(wrapper.text()).toBe('test')
  })
})
