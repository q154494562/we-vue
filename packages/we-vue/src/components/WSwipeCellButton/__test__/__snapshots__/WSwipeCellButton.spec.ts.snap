// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cell-swipe-button create 1`] = `<div class="weui-swiped-btn weui-swiped-btn_default"></div>`;
